kube_operator/client.go

```diff
@@ -208,14 +215,26 @@ func (c KubeClient) Install(tar string, envVars map[string]string, agId string)
 		if err != nil {
 			return err
 		}
-		crClient := dynClient.Resource(kindToGVRMap[newCr["kind"].(string)])
+
+		var crClients []dynamic.NamespaceableResourceInterface
+		for _, gvr := range kindToGVRMap[newCr["kind"].(string)] {
+			crClients = append(crClients, dynClient.Resource(gvr))
+		}
 
 		unstructCr := unstructured.Unstructured{Object: newCr}
 
 		// the cluster has to create the endpoint for the custom resource, this can take some time
 		glog.V(3).Infof(kwlog(fmt.Sprintf("creating operator custom resource %v", newCr)))
 		for {
-			_, err := crClient.Namespace(namespace).Create(&unstructCr, metav1.CreateOptions{})
+			for _, crClient := range crClients {
+				_, err = crClient.Namespace(namespace).Create(&unstructCr, metav1.CreateOptions{})
+				if err != nil {
+					glog.Errorf("Failed to create custom resource %v.", err)
```

Comment:
Since this error is likely to happen now that we are looping through clients for every version, let's change to a Warning and add the resource name and version to the message. Maybe we also need a success message for the one that works.

Reply:
The error is now a warning and there is a sucess message. Also the redundancy if version and versions fields are both specified has been removed